### PR TITLE
build: Upgrade the go-mod-core-contracts lib

### LINKED ISSUE #803: Upgrade go-mod-core-contracts lib

### PR DESCRIPTION
**Pull request by weichou1229**
Close #803

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/master/.github/CONTRIBUTING.md

## What is the new behavior?
Upgrade the go-mod-core-contracts lib

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by weichou1229** [comment]
Hold this PR util https://github.com/edgexfoundry/go-mod-core-contracts/pull/529 merged

**Comment by weichou1229** [comment]
Upgrade the core-contracts lib to latest.

File: internal/clients/init.go [hunk 1]
```diff
@@ -120,26 +120,26 @@ func checkServiceAvailable(ctx context.Context, serviceId string, startupTimer s
...
+				if checkServiceAvailableByPing(serviceKey, configuration, lc) == nil {
 					return true
 				}
 			}
 			startupTimer.SleepForInterval()
 		}
 	}
 
-	lc.Error(fmt.Sprintf("dependency %s service checking time out", serviceId))
+	lc.Error(fmt.Sprintf("dependency %s service checking time out", serviceKey))
```
Comment (hahattan) [inline internal/clients/init.go:136]: use `lc.Errorf`

File: pkg/models/commandvalue.go [hunk 2]
```diff
@@ -27,7 +26,8 @@ const (
 	MaxBinaryBytes = 16777216
 	// DefaultFoloatEncoding indicates the representation of floating value of reading.
 	// It would be configurable in system level in the future
-	DefaultFloatEncoding = models.Base64Encoding
+	DefaultFloatEncoding = "Base64"
+	ENotation            = "eNotation"
```
Comment (hahattan) [inline pkg/models/commandvalue.go:30]: any specific reason to make this change ?

**Review by hahattan — CHANGES_REQUESTED** [review]

File: pkg/models/commandvalue.go (hunk 2 above)
Comment (cloudxxx8) [reply pkg/models/commandvalue.go:30]: We should totally remove this part.  There is no base64 encoding anymore

File: internal/clients/init.go (hunk 1 above)
Comment (weichou1229) [reply internal/clients/init.go]: Modified.

File: pkg/models/commandvalue.go (hunk 2 above)
Comment (weichou1229) [reply pkg/models/commandvalue.go]: Modified and upgrade dependency lib.

**Review by hahattan — DISMISSED** [review]
LGTM

**Review by hahattan — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
@iain-anderson @lenny-intel for awareness

**Pull request merged**